mypy/build.py

```diff
@@ -199,26 +199,44 @@ def default_flush_errors(new_messages: List[str], is_serious: bool) -> None:
 
 
 @functools.lru_cache(maxsize=None)
-def _get_site_packages_dirs(python_executable: Optional[str]) -> List[str]:
+def _get_site_packages_dirs(python_executable: Optional[str],
+                            fscache: FileSystemCache) -> List[str]:
     """Find package directories for given python.
 
     This runs a subprocess call, which generates a list of the site package directories.
     To avoid repeatedly calling a subprocess (which can be slow!) we lru_cache the results."""
+    def make_abspath(path: str, root: str) -> str:
+        """Take a path and make it absolute relative to root if not already absolute."""
+        path = os.path.normpath(path)
+        if os.path.abspath(path) == path:
+            return path
+        else:
+            return os.path.join(root, path)
```

Comment:
This feels a bit complicated (perhaps in part because you're assigning to a parameter). Is the intention this?
```
if os.path.isabs(path):
    return os.path.normpath(path)
else:
    return os.path.join(root, os.path.normpath(path))
```
While that looks like more code the flow feels a bit clearer. Also it's possible that in one of the branches you don't actually care about normpath, but I can't tell. (Or possibly you might want to call it after calling join.) Finally I note that presumably root is assumed to be absolute itself, else the function name would be misleading.

Reply:
Yes, when I wrote this I didn't know isabs existed. :) But that is much better than what I was doing. I will use your suggested code.